### ISSUE TITLE
Exhaustion bar is rendered on top of hunger/saturation bar.

### ISSUE THREAD

**Issue by GamerGuy95953** [squeek502/AppleSkin#335]
I am having an issue where the exhaustion overlay is rendered on top of the hunger and saturation bars. I have attached an image of the issue and my log. I looked in the mod settings, but I didn't see anything about rendering settings besides enabling/disabling them.

![Screenshot_20241116_083512](https://github.com/user-attachments/assets/9a00dc92-0f62-4e93-a8bb-47588bfeadb4)

[https://mclo.gs/akCkDtq](https://mclo.gs/akCkDtq)

**Comment by squeek502** [comment]
Seems rather likely to be caused by some sort of mod incompatibility, but not sure which mod or how exactly this would happen.

If you'd like to help narrowing it down, you could try:
- Start the game with only AppleSkin and Fabric API installed and see if it still occurs
- If it doesn't, add back half of the mods you had installed
  + If it occurs now, then remove half of those you added back until it stops, and then add back half, etc until you find the particular mod that causes it
  + If it still doesn't occur, then check the other half in the same way mentioned above

**Comment by GamerGuy95953** [comment]
> Seems rather likely to be caused by some sort of mod incompatibility, but not sure which mod or how exactly this would happen.
> 
> If you'd like to help narrowing it down, you could try:
> 
> * Start the game with only AppleSkin and Fabric API installed and see if it still occurs
> * If it doesn't, add back half of the mods you had installed
>   
>   * If it occurs now, then remove half of those you added back until it stops, and then add back half, etc until you find the particular mod that causes it
>   * If it still doesn't occur, then check the other half in the same way as mentioned above

Thanks for the quick reply! It was the ImmediatelyFast mod. All my other mods are not causing the issue.

**Comment by squeek502** [comment]
Seems like this might be a regression in ImmediatelyFast, as this bug has happened (and was fixed) before:

https://github.com/RaphiMC/ImmediatelyFast/issues/38

**Comment by GamerGuy95953** [comment]
> Seems like this might be a regression in ImmediatelyFast, as this bug has happened (and was fixed) before:
> 
> [RaphiMC/ImmediatelyFast#38](https://github.com/RaphiMC/ImmediatelyFast/issues/38)

Yeah, I just saw it! I am not super familiar with GitHub. Should I make a new issue or comment on the existing issue for the mod?

**Comment by squeek502** [comment]
Making a new one and linking to the old one might be best way to go.

**Comment by johnjohn** [comment]
@squeek502 @GamerGuy95953 
Just chiming in with a workaround.
In the json config for ImmediatelyFast, disabling `hud_batching` fixes the rendering of the saturation bar layer.
If any ImmediatelyFast devs end up, poking their head in here, the HUD batching code may be a good place to look for fixing this on their end.
Performance impact for disabling feels minimal, imo, but I don't have hard numbers to really back that up. GL

**Comment by GamerGuy95953** [comment]
> @squeek502 @GamerGuy95953 Just chiming in with a workaround. In the json config for ImmediatelyFast, disabling `hud_batching` fixes the rendering of the saturation bar layer. If any ImmediatelyFast devs end up, poking their head in here, the HUD batching code may be a good place to look for fixing this on their end. Performance impact for disabling feels minimal, imo, but I don't have hard numbers to really back that up. GL

Thank you! This fixes the issue for now.
